src/main/scala/is/hail/io/bgen/BgenWriter.scala

```diff
@@ -0,0 +1,179 @@
+package is.hail.io.bgen
+
+import is.hail.annotations._
+import is.hail.utils._
+import is.hail.variant.{Genotype, Variant, VariantDataset}
+
+object BgenWriter {
+  val ploidy: Byte = 2
+  val phased: Byte = 0
+
+  def shortToBytesLE(bb: ArrayBuilder[Byte], i: Int) {
+    bb += (i & 0xff).toByte
+    bb += ((i >>> 8) & 0xff).toByte
+  }
+
+  def intToBytesLE(bb: ArrayBuilder[Byte], i: Int) {
+    bb += (i & 0xff).toByte
+    bb += ((i >>> 8) & 0xff).toByte
+    bb += ((i >>> 16) & 0xff).toByte
+    bb += ((i >>> 24) & 0xff).toByte
+  }
+
+  def stringToBytesWShortLength(bb: ArrayBuilder[Byte], s: String) {
+    val bytes = s.getBytes
+    shortToBytesLE(bb, bytes.length)
+    bytes.foreach(bb += _)
+  }
+
+  def stringToBytesWIntLength(bb: ArrayBuilder[Byte], s: String) {
+    val bytes = s.getBytes
+    intToBytesLE(bb, bytes.length)
+    bytes.foreach(bb += _)
+  }
+
+  def bitPack(input: Array[Int], nBitsPerProb: Int): Array[Byte] = {
```

Comment:
Looks like this is only used in the tests.  I'd move it there.  Looks dangerous, because the block reader should clearly be reusing the ArrayBuilder.